### PR TITLE
[BACKLOG-7168] - Data Services fails if SQL contains "=NULL"

### PR DESCRIPTION
**Pull request by denis8703**
Added an additional processing to the unknown of expression, which сondition is always FALSE

### CODE REVIEW

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/core/Condition.java

**Comment by brosander** [comment]
Head build failed with return code 1

**Comment by brosander** [comment]
Head build failed with return code 1

**Comment by brosander** [comment]
Head build failed with return code 1

**Comment by brosander** [comment]
@eliofreitas looks like trigger missed this new commit, kicking off manually

File: core/test-src/org/pentaho/di/core/ConditionTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Test;
+import org.pentaho.di.core.row.RowMeta;
+
+public class ConditionTest extends TestCase {
+  @Test
+  public void testCondition1() throws Exception {
+    Condition condition = new Condition();
+    condition.setNegated( true );
+    assertFalse( condition.evaluate( new RowMeta(), new Object[]{ "test" } ) );
+  }
```
Comment (mkambol) [inline core/test-src/org/pentaho/di/core/ConditionTest.java]: Hey @denis8703 - it doesn't look like your test will actually hit your new code.  Can you construct a Condition which uses a TRUE function and NOT operator?

Comment (denis8703) [inline core/test-src/org/pentaho/di/core/ConditionTest.java]: Hi @mkambol - yes, but changes need for test are in anothe repository (SQLCondition.java pdi-dataservice-plugin) https://github.com/pentaho/pdi-dataservice-plugin/pull/66, so the test will fail. I can add after the merger changes for https://github.com/pentaho/pdi-dataservice-plugin/pull/66 ?

Comment (mkambol) [inline core/test-src/org/pentaho/di/core/ConditionTest.java]: @denis8703 I just meant that you can construct a Condition right in your test which will actually hit line 378 in Condition.java.  The test you created doesn't.  
Small thing, but it's good to make sure the test actually covers what we think it does.  Also, this is sort of a personal preference thing, but I like test names which are descriptive-  something like 
  .testNegatedTrueFuncEvaluatesAsFalse()

Comment (denis8703) [inline core/test-src/org/pentaho/di/core/ConditionTest.java]: @mkambol Matt I did, please review.

**Pull request merged**